### PR TITLE
Friend api

### PR DESCRIPTION
**Pull request by flutterexcel**

### CODE REVIEW

File: client/friends/friends.controller.js [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+            .controller('friendCtrl', friendCtrl);
     function friendCtrl(friendService, socialSharing, $scope) {
         $scope.selected = null;
-        $scope.friendClick = (index) => {
-            this.selected = index;
-        }
-        $scope.invite = () => {
+        //       
+//        Meteor.call('sendEmail');
+        Meteor.call('friendlist', function(err, data) {
```
Comment (Rahul-Khera-Codes) [inline client/friends/friends.controller.js]: use es6 here

File: client/friends/friends.html [hunk 2]
```diff
@@ -2,6 +2,7 @@
    <h1 class="title" style="margin-left:1em;">Friends List</h1>
 </ion-header-bar>
 <ion-content>
+    <div ng-click="signout()">logout</div>
    <ul class="list no-border p-b">
       <li class="list-item" ng-repeat="friend in friendsList" ng-click="friendClick($index)">
```
Comment (Rahul-Khera-Codes) [inline client/friends/friends.html]: create a separate directive for friend box

File: client/friends/friendlistTemplate.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<ul class="list no-border p-b">
+    <li class="list-item" ng-repeat="friend in friendsList" ng-click="friendClick($index)">
```
Comment (rahulexcel) [inline client/friends/friendlistTemplate.html]: each individual friend should also be a seperate directive

File: client/friends/frndsDirective.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            .directive('friendList', friendList);
+    function friendList() {
+      var friend={};
+      friend.restrict='E';
+      friend.scope={
+          friendsList:'=list'
+          
+      };
+      friend.templateUrl='client/friends/friendlistTemplate.html'
+      friend.link=function(scope,element,attr){
```
Comment (Rahul-Khera-Codes) [inline client/friends/frndsDirective.js]: remove if empty

File: client/friends/frndsDirective.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+       
+      };
+      friend.transclude=true;
```
Comment (Rahul-Khera-Codes) [inline client/friends/frndsDirective.js:14]: why this?

File: client/login/login.controller.js [hunk 6]
```diff
@@ -3,12 +3,32 @@
...
-    function loginCtrl(facebookLogin, toastNotification, $state, $rootScope) {
+    function loginCtrl(facebookLogin, toastNotification, $state, $rootScope,$timeout) {
         this.fbLogin = () => {
+     
             facebookLogin.login().then((data) => {
                 $rootScope.friendsList = data.friends;
-                toastNotification.toast('Successfully Logged In');
-                $state.go('friends');
+                  Meteor.loginWithPassword(data.email, data.id, (err)=>{
+                if (err) {
```
Comment (Rahul-Khera-Codes) [inline client/login/login.controller.js:12]: code not formatted

File: server/friendlist.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// method for getting friend list 
+Meteor.methods({
+    friendlist: function() {
+        console.log('this function is for friend list');
+         var data=Meteor.user().profile.friends;
```
Comment (Rahul-Khera-Codes) [inline server/friendlist.js]: simply put return Meteor.user().profile.friends;

File: server/server.js [hunk 8]
```diff
@@ -1,63 +1,45 @@
...
-                    }
-                    Meteor.users.update(Meteor.user()._id, {
-                        $set: {
-                            'profile.friends': data.friends,
-                            'profile.geo': data.geos
-                        }
-                    });
+    user_signup: function(data) {
+        if (data.email && data.id) {
+            console.log('user signup');
```
Comment (Rahul-Khera-Codes) [inline server/server.js]: remove

File: server/server.js [hunk 9]
```diff
@@ -1,63 +1,45 @@
...
+            $set: {
+                'profile.friends': data.friends,
+                'profile.geo': data.geos
+            }
+        });
+    }
+});
 
-            //logout for single user 
-            Meteor.method({
```
Comment (Rahul-Khera-Codes) [inline server/server.js:47]: why this removed?

File: client/friends/frndsDirective.js (hunk 5 above)
Comment (flutterexcel) [inline client/friends/frndsDirective.js:14]: to access friend list from out side scope

File: client/friends/friendlistTemplate.html [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
```
Comment (Rahul-Khera-Codes) [inline client/friends/friendlistTemplate.html]: do we need this li ?

File: client/friends/friendsubDirective.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+(function() {
+    'use strict';
+    angular.module('closeness')
+            .directive('friendsubList', friendsubList);
```
Comment (Rahul-Khera-Codes) [inline client/friends/friendsubDirective.js]: rename this to 'friend'

File: client/friends/friendsubDirective.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    function friendsubList() {
+        var fsub = {};
+        fsub.restrict = 'E';
+        fsub.scope = {
+            friend: '=sublist'
+        };
+        fsub.templateUrl = 'client/friends/friendsubtemp.html'
```
Comment (Rahul-Khera-Codes) [inline client/friends/friendsubDirective.js]: rename to friend.html

File: client/friends/friendsubtemp.html [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<a herf="" class="list-left"><span class="w-40 avatar"><img src="{{friend.picture.data.url}}" alt="{{friend.name}}"> <i class="on b-white bottom"></i></span></a>
+<button class="button btn-sx pull-right invite-btn" ng-click="invite()" ng-class="{shown: $index == selected}">
+    Invite
+</button>
+<div class="list-body">
+    <div><a href="">{{friend.name}}</a></div>
+    <small class="text-muted text-ellipsis">Designer, Blogger</small>
```
Comment (Rahul-Khera-Codes) [inline client/friends/friendsubtemp.html]: what is this?

File: server/server.js (hunk 9 above)
Comment (flutterexcel) [inline server/server.js:47]: This is creating an error and there is no need for this.

Comment (deepakexcel) [inline server/server.js]: This is creating an error and there is no need for this.

File: client/friends/friendsubtemp.html (hunk 13 above)
Comment (flutterexcel) [inline client/friends/friendsubtemp.html]: That was dummy added by yuvraj

File: client/friends/friendlistTemplate.html (hunk 10 above)
Comment (flutterexcel) [inline client/friends/friendlistTemplate.html]: This design is created by yuvraj . he used themeforest design which was told by you.

**Pull request merged**